packages/devtools_app/lib/src/performance/performance_controller.dart

```diff
@@ -237,10 +224,33 @@ class PerformanceController extends DisposableController
         : PerformanceData();
   }
 
-  FutureOr<void> processAvailableEvents() async {
-    if (data == null) {
-      await _initData();
+  Future<void> _pullTraceEventsFromVmTimeline({
+    bool shouldPrimeThreadIds = false,
+  }) async {
+    final currentVmTime = await serviceManager.service.getVMTimelineMicros();
```

Comment:
need to make sure we don't have multiple simultaneous timeline event pulls at once for the case a pull took more than 2 seconds.
Use the RateLimiter class and perhaps a delay of only 200 ms to achieve that goal.
As you are now guaranteed to only have one pull at a time it would be safe to pull more frequently.
https://github.com/flutter/devtools/blob/554a0cbbda362785cab18172a3457f5de1fbbc38/packages/devtools_app/lib/src/utils.dart#L452